homeassistant/components/litterrobot/switch.py

```diff
@@ -1,76 +1,86 @@
 """Support for Litter-Robot switches."""
 from __future__ import annotations
 
-from typing import Any
+from collections.abc import Callable, Coroutine
+from dataclasses import dataclass
+from typing import Any, Generic, Union
 
-from pylitterbot import LitterRobot
+from pylitterbot import FeederRobot, LitterRobot
 
-from homeassistant.components.switch import SwitchEntity
+from homeassistant.components.switch import SwitchEntity, SwitchEntityDescription
 from homeassistant.config_entries import ConfigEntry
 from homeassistant.core import HomeAssistant
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
 
 from .const import DOMAIN
-from .entity import LitterRobotConfigEntity
+from .entity import LitterRobotConfigEntity, _RobotT
 from .hub import LitterRobotHub
 
 
-class LitterRobotNightLightModeSwitch(
-    LitterRobotConfigEntity[LitterRobot], SwitchEntity
-):
-    """Litter-Robot Night Light Mode Switch."""
-
-    @property
-    def is_on(self) -> bool | None:
-        """Return true if switch is on."""
-        if self._refresh_callback is not None:
-            return self._assumed_state
-        return self.robot.night_light_mode_enabled
-
-    @property
-    def icon(self) -> str:
-        """Return the icon."""
-        return "mdi:lightbulb-on" if self.is_on else "mdi:lightbulb-off"
+@dataclass
+class RobotSwitchEntityDescription(SwitchEntityDescription, Generic[_RobotT]):
+    """A class that describes robot switch entities."""
+
+    icons: tuple[str | None, str | None] = (None, None)
+    set_fn: Callable[
+        [_RobotT], Callable[[bool], Coroutine[Any, Any, bool]]
+    ] | None = None
```

Comment:
As with button entity, I suggest that you use a RequiredKeys mixin class for the compulsory items